Added sample script which uses connectovercdp

Ishita-Jinturkar opened the pull request:
Added a sample js script which uses connectovercdp

Commit:
Added sample script which uses connectovercdp

Commit:
.

puagarwa commented on samples/cdp-tests/connectOverCDPScript.js:
```diff
@@ -0,0 +1,35 @@
+  try {
+    console.log('🔌 Connecting to CDP server...');
+    const browser = await chromium.connectOverCDP('ENTER YOUR CDP WEBSOCKET URL HERE',
+    {headers:{'User-Agent': 'Chrome-DevTools-Protocol/1.3'}});
```
is this mandatory ?

puagarwa reviewed (changes requested):
Also add me readMe file with minimal instructions on how customer would use cdp from our service. Follow examples of playwright for guidance and reference.

Ishita-Jinturkar committed:
Add GetCdpUrl.py to retrieve WebSocket URL

This script retrieves the WebSocket URL for a remote browser using Playwright's API, handling environment variables and making asynchronous HTTP requests.

Ishita-Jinturkar committed:
Create .env.example for Playwright configuration

Add example environment configuration for Playwright service.

Ishita-Jinturkar committed:
Fetch and connect to CDP WebSocket URL

Added functionality to fetch CDP WebSocket URL from Playwright service and connect to the CDP server using it.

Ishita-Jinturkar committed:
Delete samples/cdp-tests/GetCdpUrl.py

Ishita-Jinturkar committed:
Add remote browser search functionality for Amazon

Ishita-Jinturkar committed:
Removed data from .env.example

Clear sensitive information from the example environment file.

Ishita-Jinturkar committed:
Update to Azure OpenAI for product search

Refactor to use Azure OpenAI instead of local Ollama LLM for Amazon product search.

Ishita-Jinturkar committed:
Change console.log to console.debug and update URL

Ishita-Jinturkar commented on samples/cdp-tests/connectOverCDPScript.js:
```diff
@@ -0,0 +1,62 @@
+import { chromium } from 'playwright';
+
+const WORKSPACE_ID = process.env.PLAYWRIGHT_SERVICE_WORKSPACE_ID;
+const REGION = process.env.PLAYWRIGHT_SERVICE_REGION;
```
Should we modify the script for 2 api flow to support this? First call will go to auth

Ishita-Jinturkar committed:
Refactor connection script for Playwright service

Ishita-Jinturkar committed:
Refactor Playwright service URL handling

Updated environment variable usage and added parsing for service URL.

Ishita-Jinturkar committed:
Refactor Browser-Use-Remote.py for clarity and dotenv

Updated prerequisites and usage instructions. Refactored remote browser session creation and added dotenv support.

Ishita-Jinturkar committed:
Refactor connectOverCDPScript.js to Python

Updated the script to connect to a remote browser via CDP using Python. Added prerequisites and usage instructions.

Ishita-Jinturkar committed:
Revise .env.example for Playwright and Azure OpenAI

Updated environment variable placeholders for Playwright and Azure OpenAI.

Ishita-Jinturkar committed:
Refactor Browser-Use-Remote.py for clarity and updates

Updated script usage instructions and refactored remote browser session creation.

Ishita-Jinturkar committed:
Add Playwright Service Python client implementation

This file provides a Python client for Microsoft Playwright Service, including methods to retrieve a CDP endpoint URL for connecting to remote browsers. It includes error handling and environment variable support.

Ishita-Jinturkar committed:
Add Playwright testing example with remote browser support

This script provides a testing framework using Microsoft Playwright Service to run tests on remote browsers via the Chrome DevTools Protocol. It includes context managers for easy browser and page access, example tests, and a test runner.

Ishita-Jinturkar committed:
Update script name in usage instructions

Ishita-Jinturkar committed:
Fix script name in usage instructions

Ishita-Jinturkar committed:
Add README for Python CDP samples

Added a README file with usage instructions and examples for Python CDP samples using Microsoft Playwright Service.

Ishita-Jinturkar committed:
Add requirements.txt for Python sample dependencies

Ishita-Jinturkar committed:
Add headers to CDP connection in connectOverCDPScript

Ishita-Jinturkar committed:
Add User-Agent header to CDP connection

Ishita-Jinturkar committed:
Update API URL to include browser and version parameters

Commit:
Add CDP connection samples for Python and JavaScript

Commit:
Merge and keep local CDP samples with fixes

Commit:
Add TODO comments for post-OSS redirect support cleanup

Commit:
Remove TODO prefix from comments
